### DISCUSSION TITLE
Redirect tabs

### DISCUSSION THREAD

**Discussion by mahesh11T** [zauberzeug/nicegui#1271] [Q&A] [answered]
### Question

how can we redirect to different tabs?
 
```py
 def some_logic():
      # some logic completed
      # redirect to some other tab after completion
      ui.open(ui.tab('D'))
```

**Comment by falkoschindler** [comment, marked as answer]
You can switch tabs by setting the `value` like this:

```py
with ui.tabs().classes('w-full') as tabs:
    one = ui.tab('One')
    two = ui.tab('Two')
with ui.tab_panels(tabs, value=two).classes('w-full'):
    with ui.tab_panel(one):
        ui.label('First tab')
    with ui.tab_panel(two):
        ui.label('Second tab')

ui.button('Switch to first tab', on_click=lambda: tabs.set_value(one)).classes('w-64')
```

**Reply by mahesh11T** [reply]
This is what I wanted thanks!

**Comment by rodja** [comment]
We have a [demo on how to switch tabs programatically](https://nicegui.io/documentation/tabs#switch_tabs_programmatically). Or do you think about "single-page application" kind of behaviour? I wrote something about that in https://github.com/zauberzeug/nicegui/discussions/1147#discussioncomment-6401630
